doc/source/working_with_kiwi/xml_description.rst

```diff
@@ -0,0 +1,763 @@
+   be included in the final appliance.
+
+
+The `repository` element accepts one `source` child element, which
+contains the URL to the repository in an appropriate format and the
+following optional attributes:
+
+.. FIXME: What's the advantage/use-case for type="yast2"
+
+- `type`: repository type, accepts one of the following values: `apt-deb`,
+  `apt-rpm`, `deb-dir`, `mirrors`, `red-carpet`, `rpm-dir`, `rpm-md`,
+  `slack-site`, `up2date-mirrors`, `urpmi`, `yast2`.
+  For ordinary RPM repositories use `rpm-md`, for ordinary APT repositories
+  `apt-deb`.
+
+- `imageinclude`: Specify whether this repository should be added to the
+  resulting image, defaults to false.
+
+- `imageonly`: A repository with `imageonly="true"` will not be available
+  during image build, but only in the resulting appliance. Defaults to
+  false.
+
+- `priority`: An integer priority for all packages in this repository. If
+  the same package is available in more than one repository, then the one
+  with the highest priority is used.
+
+- `alias`: Name to be used for this repository, it will appear as the
+  repository's name in the image, which is visible via ``zypper repos`` or
+  ``dnf repolist``. KIWI will construct an alias from the path in the
+  `source` child element (replacing each `/` with a `_`), if no value is
+  given.
+
+- `repository_gpgcheck`: Specify whether or not this specific repository is
+  configured to to run repository signature validation. If not set, the
+  package manager's default is used.
+
+- `package_gpgcheck`: Boolean value that specifies whether each package's
+  GPG signature will be verified. If omitted, the package manager's default
+  will be used
+
+- `components`: Distribution components used for `deb` repositories,
+  defaults to `main`.
+
+- `distribution`: Distribution name information, used for deb repositories.
+
+- `profiles`: List of profiles to which this repository applies.
+
+.. TODO: username & password are supported, but no clue what they are for?
+
+.. _xml-description-supported-supported-repository-paths:
+
+Supported repository paths
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The actual location of a repository is specified in the `source` child
+element of `repository` via its only attribute `path`. KIWI supports the
+following paths types:
+
+- `http://URL` or `https://URL` or `ftp://URL`: a URL to the repository
+  available via HTTP(s) or FTP.
+
+- `obs://$PROJECT/$REPOSITORY`: evaluates to the repository `$REPOSITORY`
+  of the project `$PROJECT` available in the Open Build Service (OBS). By
+  default KIWI will look for projects on `<build.opensuse.org>`_, but this
+  can be overridden by providing an URL to a different OBS instance as
+  follows:
+
+  .. code-block:: xml
+
+     <image schemaversion="7.0" name="JeOS-Tumbleweed">
+       <!-- snip -->
+       <repository type="rpm-md" alias="test">
+         <source path="obs://foo.org:myproject:test/Factory"/>
```

Comment:
That is actually working, I've used it to build stuff on IBS but include the repository from Virtualization:Appliances:Builder from OBS.

Reply:
mmm I don't know the details about how IBS has the openSUSE.org configuration defined and how is it working. I always thought that this was some sort of internal OBS readonly redirection inside a project called `openSUSE.org`, meaning that `openSUSE.org` just like another local project in IBS.

Note that the obs:// scheme is translated to a URL by KIWI in a different way if it is running inside OBS or not. If inside OBS it uses a local path, if not, it uses the URL defined in runtime configuration file. So in order to make `obs://` urls to resolve to some custom OBS instance I don't think there is any other option than using the runtime configuration file. If `obs://` is used inside any OBS instance out there it will be always consistent as it uses the local path provided by OBS itself.

Reply:
You're right, there is a `openSUSE.org` project on IBS! So that whole section is actually wrong. Thanks for pointing that out! I'll fix that.